File: shopping_mall_sellergoods_interface/src/main/java/com/mall/sellergoods/service/BrandService.java
package com.mall.sellergoods.service;


import com.mall.entity.PageResult;
import com.mall.model.TbBrand;

import java.util.List;

/**
 * @author : Winnie
 * @date :   2019/9/4
 * @description :  
 */
public interface BrandService {
    /**
     * 获取所有品牌列表
     * @return
     */
    List<TbBrand> findAll();
    /**
     * 返回分页列表
     * @return
     */
    PageResult findPage(int pageNum, int pageSize);
    /**
     * 增加品类
     */
    void add(TbBrand brand);
    /**
     * 修改
     */
    void update(TbBrand brand);
    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    TbBrand findOne(Long id);


}

File: shopping_mall_manage_web/src/main/resources/static/js/service/brandService.js
//服务层
app.service('brandService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../brand/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../brand/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../brand/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../brand/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../brand/update',entity );
	}
	//删除
	this.del=function(ids){
		return $http.get('../brand/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../brand/search?page='+page+"&rows="+rows, searchEntity);
	}
	//下拉列表数据
	this.selectOptionList=function(){
		return $http.get('../brand/selectOptionList');
	}
	
});
